### ISSUE TITLE
incompatible with pyarrow==0.15.0

### ISSUE THREAD

**Issue by abditag2** [uber/petastorm#437]
Pyarrow might have introduced some breaking changes in 0.15.0

```
E   Traceback (most recent call last):
E       schema_fields=schema_fields) as train_reader:
E     File "/tmp/michelangelo-python-client/.tox/py27/local/lib/python2.7/site-packages/petastorm/reader.py", line 289, in make_batch_reader
E       is_batched_reader=True)
E     File "/tmp/michelangelo-python-client/.tox/py27/local/lib/python2.7/site-packages/petastorm/reader.py", line 360, in __init__
E       stored_schema = infer_or_load_unischema(self.dataset)
E     File "/tmp/michelangelo-python-client/.tox/py27/local/lib/python2.7/site-packages/petastorm/etl/dataset_metadata.py", line 394, in infer_or_load_unischema
E       return Unischema.from_arrow_schema(dataset)
E     File "/tmp/michelangelo-python-client/.tox/py27/local/lib/python2.7/site-packages/petastorm/unischema.py", line 251, in from_arrow_schema
E       meta = parquet_dataset.pieces[0].get_metadata(parquet_dataset.fs.open)
E   TypeError: get_metadata() takes exactly 1 argument (2 given)
```

**Comment by selitvin** [comment]
Yes. I am working on a fix in this PR: #438. There are some issues with 0.15.0 support of list of elements fields which breaks some tests: https://issues.apache.org/jira/browse/ARROW-6844.

**Comment by selitvin** [comment]
0.7.7 was released. It should solve the compatibility issue.
